woocommerce/woocommerce-gutenberg-products-block
# Refactor: Style quantity input dynamically

Fixes #6029, replaces #6087

This PR refactors the style of input quantity used by Cart and Mini Cart blocks to make it dynamic from the CSS side. This PR tries to replicate as close as possible the original style of the quantity input but there are some parts changed to make it dynamic.

### Screenshots

_Notes: The original style is on the right. The style from this PR is on the left._



Default state (Cart block):
<img width="1169" alt="image" src="https://user-images.githubusercontent.com/5423135/160886629-dffbd441-44ba-4b3c-b2bc-0d8e7f9d6b56.png">

The new border is darker and bolder because it works better for colors other than black.

Focused input: (Mini Cart block)
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/5423135/160887228-7726e9b5-9ca4-423d-bcd6-51ad00614fc5.png">

Focused button: (Mini Cart block)
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/5423135/160887502-9a5b3da5-67e7-4070-9146-2ebe4d5eaffd.png">

Dynamic style: (Mini Cart block)
<img width="980" alt="image" src="https://user-images.githubusercontent.com/5423135/160888103-7e3aad9a-f150-4d67-9afc-86e16c54d49c.png">



### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Checkout this PR and build the assets.
2. Test the quantity input of the Cart block, see no issue, see style for normal, hover, focused, and disabled state of input and button look good.
3. For Mini Cart, set the text color and background of the Mini Cart Contents block. See the input style changes dynamically.
### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
